shared/src/test/java/io/crate/concurrent/CompletableFuturesTest.java

```diff
@@ -71,4 +76,23 @@ public void testAllAsListResultContainsListOfResults() throws Exception {
 
         assertThat(all.get(10, TimeUnit.SECONDS), Matchers.contains(10, 20));
     }
+
+    @Test
+    public void testSupplyAsyncReturnsFailedFutureOnException() throws Exception {
+        ExecutorService executorService = new ThreadPoolExecutor(1, 1, 1,
+            TimeUnit.MILLISECONDS, new ArrayBlockingQueue<>(1));
```

Comment:
Might be easier to define an anonymous ad-hoc `Executor` that raises an Exception immediately.